lib/astutils.cpp

```diff
@@ -147,11 +147,20 @@ static bool isInLoopCondition(const Token * tok)
     return Token::Match(tok->astTop()->previous(), "for|while (");
 }
 
+/// If tok2 comes after tok1
+static bool precedes(const Token * tok1, const Token * tok2)
+{
+    if(!tok1)
+        return false;
+    if(!tok2)
+        return false;
+    return std::make_pair(tok1->linenr(), tok1->col()) < std::make_pair(tok2->linenr(), tok2->col());
```

Comment:
I did not know we could use `std::make_pair` like this. Does the first member have precedence?

Reply:
I am not 100% sure that the line,col always "increase" in the token list after preprocessor expansions and simplifications. The `Token::progressValue()` is another option. It was many years since I used it but as far as I remember, after all simplifications that should always increase when you loop forward in the token list.

Reply:
in general this `precedes` function will not work because it does not care what file the tokens come from. But maybe that is not a problem when it's used from `followVariableExpressions`.

Reply:
> Does the first member have precedence?

Yes, it does a lexicographical compare.

> The Token::progressValue() is another option.

Let me try that. I didn't know what that was for.

> in general this precedes function will not work because it does not care what file the tokens come from.

We should only call followVar when it is in the same file. Let me add a check for that.